src/ui/molecules/networkGraph/ecs/system/nodeLabelSystem.ts

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+/* eslint-disable @typescript-eslint/prefer-readonly-parameter-types */
+import type { ComponentOf } from '@javelin/ecs'
+import { component, createQuery, useMonitor } from '@javelin/ecs'
+import type { Node } from 'ngraph.graph'
+import { LabelComponent, NodeComponent } from '../schema'
+import type { SystemFactory, World, WorldTickData } from '../type'
+
+const queries = {
+  nodes: createQuery(NodeComponent)
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type LabelExtractor<T = any> = (node: Node<T>) => string | null
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type NodeLabelSystemOptions<T = any> = {
+  labelExtractor: LabelExtractor<T>
+}
+
+export const defaultLabelExtractor = (node: Node<Record<string, string> | null>): string | null => {
+  return node.data?.['label'] ?? null
```

Comment:
The optional chaining returns `undefined` not `null` so the nullish coalescing operator is necessary to satisfy the union type `string | null`.

But ok for the unnecessary <sub>(ugly)</sub> `return`. It will be removed. 😌

Reply:
Yep, of course it returns `undefined`, my bad 😌